### DISCUSSION TITLE
Lineage problem (need to cleanup or reset lineages)

### DISCUSSION THREAD

**Discussion by mchristianto** [open-metadata/OpenMetadata#17182] [Q&A]
First, I ingest Big Query lineages without filtering user_email to specific service account (that run production queries), so i think Openmetadata ingest all the queries executed by all users. 

All the lineages is created successfully but it has wrong nodes and edges, also the query lineage show the wrong query, because it shows create table as select (CTAS) query statement from the old table that even didn't exists anymore in Openmetadata.

I edit the lineage ingestion config and added filter user_email so it just scanned query history executed by specific service account, the ingestion is success but when i see the lineage its not changing, still with the wrong sql query lineage.

I think the first lineage cannot be replaced by the new one with filter, is there any way to delete all lineage, reset or cleanup so i can ingest fresh lineage from Big Query?

**Comment by Prajwal214** [comment]
To reset or clean up the existing lineage in OpenMetadata and ingest fresh lineage from BigQuery, you need to manually delete the outdated lineage edges before re-running the ingestion job. Here are the steps:

1. Delete Existing Lineage: 
            a. Navigate to the Lineage Tab: Go to the data asset's details page and click on the "Lineage" tab.
            b. Edit Lineage: Click on the "Edit" icon to enter the lineage editor mode.
            c. Delete Edges: Select the edge (line) you want to delete and click on the delete option to remove it.
            
<img width="728" alt="image" src="https://github.com/user-attachments/assets/22172e06-d542-4625-a866-4573ce034d00">

2. Modify and Re-run Ingestion: After deleting the outdated edges, re-run the ingestion job with the updated configuration that includes the filter for user_email.

**Reply by mchristianto** [reply]
Thanks for your answer, but i think i faced strange condition here, my lineage is somehow like this:

![image](https://github.com/user-attachments/assets/f810eda6-e552-4af2-8630-564a490cfb3f)

little edge on the left its like bug or something, and its affect when i click the hide (-) button like this:

![image](https://github.com/user-attachments/assets/40c8b204-2762-4744-8ee2-ee9109ed13b2)

![image](https://github.com/user-attachments/assets/ab0058ed-5b0a-4abe-a813-a339ba0f182f)

it cannot expanded anymore, idk because it still showing hide (-) button, and on the right most we can see little edge beside the hide (-) button. I need to refresh the page to got to the initial condition.

I cannot delete the little edges and of course cannot delete the main node, that's because i want to cleanup all of the lineage and ingest the fresh lineage, it happens in almost every lineage.

i think the only option now is to delete all metadata and fresh ingest the metadata, after that we fresh ingest the lineage....but my table or view already filled with description, domain, owner, etc.. i cannot delete entire object like that. Is there anything we can do to overcome this?

**Reply by Prajwal214** [reply]
@mchristianto Can you share the OM Version?

**Reply by Prajwal214** [reply]
You need to click on edit Lineage, And then you will get the cross button to delete the node. You can run the ingestion once the Lineage node is deleted. 
<img width="376" alt="image" src="https://github.com/user-attachments/assets/c44220ce-0470-46b4-9e48-8850b38e4b4d">

**Reply by mchristianto** [reply]
i'm using version 1.4.5

**Reply by mchristianto** [reply]
i already click on the edit mode, if its "full edge" i can remove it

![image](https://github.com/user-attachments/assets/e268e2bd-2045-4384-83ee-a0213da0dc88)

but there are an edge with no connection to another nodes, it cannot be remove 

![image](https://github.com/user-attachments/assets/94b1ee83-903f-4e45-b179-1473f1c5cc93)
 
there is no edit edge option when i clicked on it, so i cannot remove the edges on both side

**Reply by Prajwal214** [reply]
Can you try deleting the edge using the API,
https://docs.open-metadata.org/swagger.html#operation/deleteLineageEdge
